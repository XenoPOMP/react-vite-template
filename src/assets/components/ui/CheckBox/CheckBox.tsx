import { PropsWith } from '@xenopomp/advanced-types';

import cn from 'classnames';
import { FC, PropsWithChildren, useRef } from 'react';

import styles from './CheckBox.module.scss';
import type { CheckBoxProps } from './CheckBox.props';

/**
 * Check box component. Built with native checkbox element.
 *
 * @example
 * const [checked, setChecked] = useState<boolean>(false);
 *
 * <CheckBox checked={checked} onClick={state => setChecked(state)}>
 *   Usual
 * </CheckBox>
 *
 * @param children
 * @param checked				determines whether checkbox is
 * 											checked or not.
 * @param disabled			disables checkbox.
 * @param onClick				click callback.
 * @constructor
 */
const CheckBox: FC<PropsWith<'children', CheckBoxProps>> = ({
  children,
  checked,
  disabled,
  onClick,
}) => {
  return (
    <label className={cn(styles.checkBoxLabel)}>
      <input
        type={'checkbox'}
        checked={checked}
        disabled={disabled}
        className={cn(styles.boxInput)}
        onChange={event => {
          if (onClick) {
            onClick(event.target.checked);
          }
        }}
      />
      <span className={cn(styles.mark)}>
        {!checked && !disabled && (
          <svg
            viewBox='0 0 18 18'
            fill='none'
            xmlns='http://www.w3.org/2000/svg'
          >
            <rect
              x='1.5'
              y='1.5'
              width='15'
              height='15'
              rx='2.5'
              stroke='#737674'
            />
            <rect
              className={cn(styles.active)}
              x='1.5'
              y='1.5'
              width='15'
              height='15'
              rx='2.5'
              stroke='#0066FF'
            />
          </svg>
        )}

        {checked && !disabled && (
          <svg
            viewBox='0 0 18 18'
            fill='none'
            xmlns='http://www.w3.org/2000/svg'
          >
            <rect
              className={cn(styles.fillActive)}
              x='1.5'
              y='1.5'
              width='15'
              height='15'
              rx='2.5'
              fill='#0066FF'
            />
            <rect
              x='1.5'
              y='1.5'
              width='15'
              height='15'
              rx='2.5'
              stroke='#737674'
            />
            <rect
              className={cn(styles.active)}
              x='1.5'
              y='1.5'
              width='15'
              height='15'
              rx='2.5'
              stroke='#0066FF'
            />
            <path
              className={cn(styles.markIcon)}
              d='M4.5 7.00004L9.00001 11.5L7.50001 13.0001L3 8.50004L4.5 7.00004Z'
              fill='white'
            />
            <path
              className={cn(styles.markIcon)}
              d='M15 5.5L9.00001 11.5L7.49999 10L13.5 4L15 5.5Z'
              fill='white'
            />
          </svg>
        )}

        {!checked && disabled && (
          <svg
            viewBox='0 0 18 18'
            fill='none'
            xmlns='http://www.w3.org/2000/svg'
          >
            <rect
              className={cn(styles.notActive)}
              x='1.5'
              y='1.5'
              width='15'
              height='15'
              rx='2.5'
              stroke='#737674'
            />
          </svg>
        )}

        {checked && disabled && (
          <svg
            viewBox='0 0 18 18'
            fill='none'
            xmlns='http://www.w3.org/2000/svg'
          >
            <rect
              className={cn(styles.notActive, styles.fillNotActive)}
              x='1.5'
              y='1.5'
              width='15'
              height='15'
              rx='2.5'
              fill='#737674'
              stroke='#737674'
            />
            <path
              className={cn(styles.markIcon)}
              d='M4.5 7.00004L9.00001 11.5L7.50001 13.0001L3 8.50004L4.5 7.00004Z'
              fill='white'
            />
            <path
              className={cn(styles.markIcon)}
              d='M15 5.5L9.00001 11.5L7.49999 10L13.5 4L15 5.5Z'
              fill='white'
            />
          </svg>
        )}
      </span>

      {children}
    </label>
  );
};

export default CheckBox;
