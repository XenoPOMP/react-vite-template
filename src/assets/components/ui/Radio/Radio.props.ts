export interface RadioProps {
  name?: string;
}
