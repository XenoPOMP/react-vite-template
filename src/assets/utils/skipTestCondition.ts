import useEnv, { TestingMode } from '@hooks/useEnv';

const { TESTING_MODE } = useEnv();

/**
 * Determines skip test condition.
 * @param {TestingMode} mode       target mode
 * @returns {boolean}              false if current mode isn`t the same as target one
 *                                 or current mode is FULLSTACK
 */
const skipTestCondition = (mode: TestingMode): boolean => {
  if (TESTING_MODE === 'FULLSTACK') {
    return false;
  }

  return TESTING_MODE !== mode;
};

export { skipTestCondition as default };
