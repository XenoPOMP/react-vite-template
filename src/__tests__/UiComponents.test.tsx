import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import CheckBox from '@ui/CheckBox/CheckBox';
import Radio from '@ui/Radio/Radio';

import skipTestCondition from '@utils/skipTestCondition';

import { defaultTestingPageRender } from './defaults';

describe.skipIf(skipTestCondition('FRONTEND'))('UI Components', () => {
  test('Match snapshot', () => {
    expect(render(<CheckBox>It is checkbox!</CheckBox>)).toMatchSnapshot();
    expect(
      render(<CheckBox checked>It is checkbox!</CheckBox>)
    ).toMatchSnapshot();
    expect(
      render(<CheckBox disabled>It is checkbox!</CheckBox>)
    ).toMatchSnapshot();
    expect(
      render(
        <CheckBox checked disabled>
          It is checkbox!
        </CheckBox>
      )
    ).toMatchSnapshot();

    // expect(render(<CustomSelect />)).toMatchSnapshot();
    expect(render(<Radio />)).toMatchSnapshot();
  });

  test('Checkbox click event', () => {
    const page = defaultTestingPageRender().asFragment();

    const textField = page.querySelector('#check-preview');
    const text = textField !== null ? textField.textContent : 'false';
    const button = page.querySelector('button#checkbox-test');

    expect(text).toBe('false');
  });
});
