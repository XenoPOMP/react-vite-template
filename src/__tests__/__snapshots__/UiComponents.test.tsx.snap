// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UI Components > Match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="_checkBoxLabel_05fc2d"
      >
        <input
          class="_boxInput_05fc2d"
          type="checkbox"
        />
        <span
          class="_mark_05fc2d"
        >
          <svg
            fill="none"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="15"
              rx="2.5"
              stroke="#737674"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              class="_active_05fc2d"
              height="15"
              rx="2.5"
              stroke="#0066FF"
              width="15"
              x="1.5"
              y="1.5"
            />
          </svg>
        </span>
        It is checkbox!
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="_checkBoxLabel_05fc2d"
    >
      <input
        class="_boxInput_05fc2d"
        type="checkbox"
      />
      <span
        class="_mark_05fc2d"
      >
        <svg
          fill="none"
          viewBox="0 0 18 18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="15"
            rx="2.5"
            stroke="#737674"
            width="15"
            x="1.5"
            y="1.5"
          />
          <rect
            class="_active_05fc2d"
            height="15"
            rx="2.5"
            stroke="#0066FF"
            width="15"
            x="1.5"
            y="1.5"
          />
        </svg>
      </span>
      It is checkbox!
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI Components > Match snapshot 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="_checkBoxLabel_05fc2d"
      >
        <input
          class="_boxInput_05fc2d"
          type="checkbox"
        />
        <span
          class="_mark_05fc2d"
        >
          <svg
            fill="none"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="15"
              rx="2.5"
              stroke="#737674"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              class="_active_05fc2d"
              height="15"
              rx="2.5"
              stroke="#0066FF"
              width="15"
              x="1.5"
              y="1.5"
            />
          </svg>
        </span>
        It is checkbox!
      </label>
    </div>
    <div>
      <label
        class="_checkBoxLabel_05fc2d"
      >
        <input
          checked=""
          class="_boxInput_05fc2d"
          type="checkbox"
        />
        <span
          class="_mark_05fc2d"
        >
          <svg
            fill="none"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              class="_fillActive_05fc2d"
              fill="#0066FF"
              height="15"
              rx="2.5"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              height="15"
              rx="2.5"
              stroke="#737674"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              class="_active_05fc2d"
              height="15"
              rx="2.5"
              stroke="#0066FF"
              width="15"
              x="1.5"
              y="1.5"
            />
            <path
              class="_markIcon_05fc2d"
              d="M4.5 7.00004L9.00001 11.5L7.50001 13.0001L3 8.50004L4.5 7.00004Z"
              fill="white"
            />
            <path
              class="_markIcon_05fc2d"
              d="M15 5.5L9.00001 11.5L7.49999 10L13.5 4L15 5.5Z"
              fill="white"
            />
          </svg>
        </span>
        It is checkbox!
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="_checkBoxLabel_05fc2d"
    >
      <input
        checked=""
        class="_boxInput_05fc2d"
        type="checkbox"
      />
      <span
        class="_mark_05fc2d"
      >
        <svg
          fill="none"
          viewBox="0 0 18 18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            class="_fillActive_05fc2d"
            fill="#0066FF"
            height="15"
            rx="2.5"
            width="15"
            x="1.5"
            y="1.5"
          />
          <rect
            height="15"
            rx="2.5"
            stroke="#737674"
            width="15"
            x="1.5"
            y="1.5"
          />
          <rect
            class="_active_05fc2d"
            height="15"
            rx="2.5"
            stroke="#0066FF"
            width="15"
            x="1.5"
            y="1.5"
          />
          <path
            class="_markIcon_05fc2d"
            d="M4.5 7.00004L9.00001 11.5L7.50001 13.0001L3 8.50004L4.5 7.00004Z"
            fill="white"
          />
          <path
            class="_markIcon_05fc2d"
            d="M15 5.5L9.00001 11.5L7.49999 10L13.5 4L15 5.5Z"
            fill="white"
          />
        </svg>
      </span>
      It is checkbox!
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI Components > Match snapshot 3`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="_checkBoxLabel_05fc2d"
      >
        <input
          class="_boxInput_05fc2d"
          type="checkbox"
        />
        <span
          class="_mark_05fc2d"
        >
          <svg
            fill="none"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="15"
              rx="2.5"
              stroke="#737674"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              class="_active_05fc2d"
              height="15"
              rx="2.5"
              stroke="#0066FF"
              width="15"
              x="1.5"
              y="1.5"
            />
          </svg>
        </span>
        It is checkbox!
      </label>
    </div>
    <div>
      <label
        class="_checkBoxLabel_05fc2d"
      >
        <input
          checked=""
          class="_boxInput_05fc2d"
          type="checkbox"
        />
        <span
          class="_mark_05fc2d"
        >
          <svg
            fill="none"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              class="_fillActive_05fc2d"
              fill="#0066FF"
              height="15"
              rx="2.5"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              height="15"
              rx="2.5"
              stroke="#737674"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              class="_active_05fc2d"
              height="15"
              rx="2.5"
              stroke="#0066FF"
              width="15"
              x="1.5"
              y="1.5"
            />
            <path
              class="_markIcon_05fc2d"
              d="M4.5 7.00004L9.00001 11.5L7.50001 13.0001L3 8.50004L4.5 7.00004Z"
              fill="white"
            />
            <path
              class="_markIcon_05fc2d"
              d="M15 5.5L9.00001 11.5L7.49999 10L13.5 4L15 5.5Z"
              fill="white"
            />
          </svg>
        </span>
        It is checkbox!
      </label>
    </div>
    <div>
      <label
        class="_checkBoxLabel_05fc2d"
      >
        <input
          class="_boxInput_05fc2d"
          disabled=""
          type="checkbox"
        />
        <span
          class="_mark_05fc2d"
        >
          <svg
            fill="none"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              class="_notActive_05fc2d"
              height="15"
              rx="2.5"
              stroke="#737674"
              width="15"
              x="1.5"
              y="1.5"
            />
          </svg>
        </span>
        It is checkbox!
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="_checkBoxLabel_05fc2d"
    >
      <input
        class="_boxInput_05fc2d"
        disabled=""
        type="checkbox"
      />
      <span
        class="_mark_05fc2d"
      >
        <svg
          fill="none"
          viewBox="0 0 18 18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            class="_notActive_05fc2d"
            height="15"
            rx="2.5"
            stroke="#737674"
            width="15"
            x="1.5"
            y="1.5"
          />
        </svg>
      </span>
      It is checkbox!
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI Components > Match snapshot 4`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="_checkBoxLabel_05fc2d"
      >
        <input
          class="_boxInput_05fc2d"
          type="checkbox"
        />
        <span
          class="_mark_05fc2d"
        >
          <svg
            fill="none"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="15"
              rx="2.5"
              stroke="#737674"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              class="_active_05fc2d"
              height="15"
              rx="2.5"
              stroke="#0066FF"
              width="15"
              x="1.5"
              y="1.5"
            />
          </svg>
        </span>
        It is checkbox!
      </label>
    </div>
    <div>
      <label
        class="_checkBoxLabel_05fc2d"
      >
        <input
          checked=""
          class="_boxInput_05fc2d"
          type="checkbox"
        />
        <span
          class="_mark_05fc2d"
        >
          <svg
            fill="none"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              class="_fillActive_05fc2d"
              fill="#0066FF"
              height="15"
              rx="2.5"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              height="15"
              rx="2.5"
              stroke="#737674"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              class="_active_05fc2d"
              height="15"
              rx="2.5"
              stroke="#0066FF"
              width="15"
              x="1.5"
              y="1.5"
            />
            <path
              class="_markIcon_05fc2d"
              d="M4.5 7.00004L9.00001 11.5L7.50001 13.0001L3 8.50004L4.5 7.00004Z"
              fill="white"
            />
            <path
              class="_markIcon_05fc2d"
              d="M15 5.5L9.00001 11.5L7.49999 10L13.5 4L15 5.5Z"
              fill="white"
            />
          </svg>
        </span>
        It is checkbox!
      </label>
    </div>
    <div>
      <label
        class="_checkBoxLabel_05fc2d"
      >
        <input
          class="_boxInput_05fc2d"
          disabled=""
          type="checkbox"
        />
        <span
          class="_mark_05fc2d"
        >
          <svg
            fill="none"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              class="_notActive_05fc2d"
              height="15"
              rx="2.5"
              stroke="#737674"
              width="15"
              x="1.5"
              y="1.5"
            />
          </svg>
        </span>
        It is checkbox!
      </label>
    </div>
    <div>
      <label
        class="_checkBoxLabel_05fc2d"
      >
        <input
          checked=""
          class="_boxInput_05fc2d"
          disabled=""
          type="checkbox"
        />
        <span
          class="_mark_05fc2d"
        >
          <svg
            fill="none"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              class="_notActive_05fc2d _fillNotActive_05fc2d"
              fill="#737674"
              height="15"
              rx="2.5"
              stroke="#737674"
              width="15"
              x="1.5"
              y="1.5"
            />
            <path
              class="_markIcon_05fc2d"
              d="M4.5 7.00004L9.00001 11.5L7.50001 13.0001L3 8.50004L4.5 7.00004Z"
              fill="white"
            />
            <path
              class="_markIcon_05fc2d"
              d="M15 5.5L9.00001 11.5L7.49999 10L13.5 4L15 5.5Z"
              fill="white"
            />
          </svg>
        </span>
        It is checkbox!
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="_checkBoxLabel_05fc2d"
    >
      <input
        checked=""
        class="_boxInput_05fc2d"
        disabled=""
        type="checkbox"
      />
      <span
        class="_mark_05fc2d"
      >
        <svg
          fill="none"
          viewBox="0 0 18 18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            class="_notActive_05fc2d _fillNotActive_05fc2d"
            fill="#737674"
            height="15"
            rx="2.5"
            stroke="#737674"
            width="15"
            x="1.5"
            y="1.5"
          />
          <path
            class="_markIcon_05fc2d"
            d="M4.5 7.00004L9.00001 11.5L7.50001 13.0001L3 8.50004L4.5 7.00004Z"
            fill="white"
          />
          <path
            class="_markIcon_05fc2d"
            d="M15 5.5L9.00001 11.5L7.49999 10L13.5 4L15 5.5Z"
            fill="white"
          />
        </svg>
      </span>
      It is checkbox!
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI Components > Match snapshot 5`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="_checkBoxLabel_05fc2d"
      >
        <input
          class="_boxInput_05fc2d"
          type="checkbox"
        />
        <span
          class="_mark_05fc2d"
        >
          <svg
            fill="none"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="15"
              rx="2.5"
              stroke="#737674"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              class="_active_05fc2d"
              height="15"
              rx="2.5"
              stroke="#0066FF"
              width="15"
              x="1.5"
              y="1.5"
            />
          </svg>
        </span>
        It is checkbox!
      </label>
    </div>
    <div>
      <label
        class="_checkBoxLabel_05fc2d"
      >
        <input
          checked=""
          class="_boxInput_05fc2d"
          type="checkbox"
        />
        <span
          class="_mark_05fc2d"
        >
          <svg
            fill="none"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              class="_fillActive_05fc2d"
              fill="#0066FF"
              height="15"
              rx="2.5"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              height="15"
              rx="2.5"
              stroke="#737674"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              class="_active_05fc2d"
              height="15"
              rx="2.5"
              stroke="#0066FF"
              width="15"
              x="1.5"
              y="1.5"
            />
            <path
              class="_markIcon_05fc2d"
              d="M4.5 7.00004L9.00001 11.5L7.50001 13.0001L3 8.50004L4.5 7.00004Z"
              fill="white"
            />
            <path
              class="_markIcon_05fc2d"
              d="M15 5.5L9.00001 11.5L7.49999 10L13.5 4L15 5.5Z"
              fill="white"
            />
          </svg>
        </span>
        It is checkbox!
      </label>
    </div>
    <div>
      <label
        class="_checkBoxLabel_05fc2d"
      >
        <input
          class="_boxInput_05fc2d"
          disabled=""
          type="checkbox"
        />
        <span
          class="_mark_05fc2d"
        >
          <svg
            fill="none"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              class="_notActive_05fc2d"
              height="15"
              rx="2.5"
              stroke="#737674"
              width="15"
              x="1.5"
              y="1.5"
            />
          </svg>
        </span>
        It is checkbox!
      </label>
    </div>
    <div>
      <label
        class="_checkBoxLabel_05fc2d"
      >
        <input
          checked=""
          class="_boxInput_05fc2d"
          disabled=""
          type="checkbox"
        />
        <span
          class="_mark_05fc2d"
        >
          <svg
            fill="none"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              class="_notActive_05fc2d _fillNotActive_05fc2d"
              fill="#737674"
              height="15"
              rx="2.5"
              stroke="#737674"
              width="15"
              x="1.5"
              y="1.5"
            />
            <path
              class="_markIcon_05fc2d"
              d="M4.5 7.00004L9.00001 11.5L7.50001 13.0001L3 8.50004L4.5 7.00004Z"
              fill="white"
            />
            <path
              class="_markIcon_05fc2d"
              d="M15 5.5L9.00001 11.5L7.49999 10L13.5 4L15 5.5Z"
              fill="white"
            />
          </svg>
        </span>
        It is checkbox!
      </label>
    </div>
    <div>
      <label
        class="_radio_db5d86"
      >
        <input
          class="_radioButton_db5d86"
          type="radio"
        />
        <span
          class="_mark_db5d86"
        >
          <svg
            class="_unChecked_db5d86"
            fill="none"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="15"
              rx="7.5"
              stroke="#737674"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              class="_outline_db5d86"
              height="15"
              rx="7.5"
              stroke="#0066FF"
              width="15"
              x="1.5"
              y="1.5"
            />
          </svg>
          <svg
            class="_checked_db5d86"
            fill="none"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              class="_fill_db5d86"
              fill="#0066FF"
              height="15"
              rx="7.5"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              height="15"
              rx="7.5"
              stroke="#737674"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              class="_outline_db5d86"
              height="15"
              rx="7.5"
              stroke="#0066FF"
              width="15"
              x="1.5"
              y="1.5"
            />
            <rect
              class="_icon_db5d86"
              fill="white"
              height="8"
              rx="4"
              width="8"
              x="5"
              y="5"
            />
          </svg>
        </span>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="_radio_db5d86"
    >
      <input
        class="_radioButton_db5d86"
        type="radio"
      />
      <span
        class="_mark_db5d86"
      >
        <svg
          class="_unChecked_db5d86"
          fill="none"
          viewBox="0 0 18 18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="15"
            rx="7.5"
            stroke="#737674"
            width="15"
            x="1.5"
            y="1.5"
          />
          <rect
            class="_outline_db5d86"
            height="15"
            rx="7.5"
            stroke="#0066FF"
            width="15"
            x="1.5"
            y="1.5"
          />
        </svg>
        <svg
          class="_checked_db5d86"
          fill="none"
          viewBox="0 0 18 18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            class="_fill_db5d86"
            fill="#0066FF"
            height="15"
            rx="7.5"
            width="15"
            x="1.5"
            y="1.5"
          />
          <rect
            height="15"
            rx="7.5"
            stroke="#737674"
            width="15"
            x="1.5"
            y="1.5"
          />
          <rect
            class="_outline_db5d86"
            height="15"
            rx="7.5"
            stroke="#0066FF"
            width="15"
            x="1.5"
            y="1.5"
          />
          <rect
            class="_icon_db5d86"
            fill="white"
            height="8"
            rx="4"
            width="8"
            x="5"
            y="5"
          />
        </svg>
      </span>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
